src/node/node_state.h

```diff
@@ -966,6 +969,20 @@ namespace ccf
       std::lock_guard<SpinLock> guard(lock);
       sm.expect(State::partOfNetwork);
 
+      // Because submitted recovery shares are encrypted with the latest ledger
+      // secret, it is not possible to rekey the ledger if the service is in
+      // that state.
+      GenesisGenerator g(network, tx);
+      if (
+        g.get_service_status().value() ==
+        ServiceStatus::WAITING_FOR_RECOVERY_SHARES)
+      {
+        LOG_FAIL_FMT(
+          "Ledger could not be rekeyed while the service is waiting for "
```

Comment:
Likewise I would prefer "Cannot ...", which implies pre-condition validation rather than "could not" which sounds like an after-the-fact error. Maybe I'm being pedantic, deferring to @eddyashton.